chainercv/utils/bbox/non_maximum_suppression.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import division
+import numpy as np
+
+from chainer import cuda
+
+from chainercv.utils.bbox._nms_gpu_post import _nms_gpu_post
+
+
+if cuda.available:
+    import cupy
+
+    @cupy.util.memoize(for_each_device=True)
+    def _load_kernel(kernel_name, code, options=()):
+        assert isinstance(options, tuple)
+        kernel_code = cupy.cuda.compile_with_cache(code, options=options)
+        return kernel_code.get_function(kernel_name)
+
+
+def non_maximum_suppression(bbox, thresh, score=None,
+                            limit=None):
+    """Suppress bounding boxes according to their Jaccard overlap.
+
+    This method checks each bounding box sequentially and selects the bounding
+    box if the Jaccard overlap between the bounding box and the previously
+    selected bounding boxes is less than :obj:`thresh`. This method is
+    mainly used as postprocessing of object detection.
+    The bounding boxes are selected from ones with higher scores.
+    If :obj:`score` is not provided as an argument, the bounding box
+    is ordered by its index in ascending order.
+
+    The bounding boxes are expected to be packed into a two dimensional
+    tensor of shape :math:`(R, 4)`, where :math:`R` is the number of
+    bounding boxes in the image. The second axis represents attributes of
+    the bounding box. They are :obj:`(x_min, y_min, x_max, y_max)`,
+    where the four attributes are coordinates of the bottom left and the
+    top right vertices.
+
+    :obj:`score` is a float array of shape :math:`(R,)`. Each score indicates
+    confidence of prediction.
+
+    This function accepts both :obj:`numpy.ndarray` and :obj:`cupy.ndarray` as
+    inputs. Please note that both :obj:`bbox` and :obj:`score` need to be
+    same type.
+    The output is same type as the type of the inputs.
+
+    Args:
+        bbox (array): Bounding boxes to be transformed. The shape is
+            :math:`(R, 4)`. :math:`R` is the number of bounding boxes.
+        thresh (float): Thresold of Jaccard overlap.
+        score (array): An array of confidences whose shape is :math:`(R,)`.
+        limit (int): The upper bound of the number of the output bounding
+            boxes. If it is not specified, this method selects as many
+            bounding boxes as possible.
+
+    Returns:
+        array:
+        An array with indices of bounding boxes that are selected. \
+        The shape of this array is :math:`(R,)` and its dtype is\
```

Comment:
:math:`(R,)` looks the length of the output is same as that of input. How about `(K,)` or `(R',)`? It is better to note `R' <= R`.